packages/react-native-sdk/src/components/PermissionsNotification.tsx

```diff
@@ -0,0 +1,109 @@
+import { AxiosError, OwnCapability } from '@stream-io/video-client';
+import { useCall, useHasPermissions } from '@stream-io/video-react-bindings';
+import {
+  PropsWithChildren,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { Alert, Pressable } from 'react-native';
+
+export type PermissionNotificationProps = PropsWithChildren<{
+  /**
+   * The permission to check for.
+   */
+  permission: OwnCapability;
+
+  /**
+   * The message to display in the notification once
+   * the requested permission is granted.
+   */
+  messageApproved: string;
+
+  /**
+   * The message to display in the notification once a permission
+   * is revoked.
+   */
+  messageRevoked: string;
+
+  /**
+   * The message to display in the notification while
+   * the requested permission is awaiting approval.
+   */
+  messageAwaitingApproval: string;
+}>;
+
+export const PermissionNotification = (props: PermissionNotificationProps) => {
```

Comment:
Very difficult to read component, not sure about it's implem. let's speak about it

Reply:
We discussed this component, and we agreed that this approach is not readable. We will refactor this one and take the 'check before action' approach, I.E 

check whether use may toggle video
if yes continue, if not send request